apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -110,16 +110,42 @@ const getBlocksTracker = ( eventName ) => ( blockIds ) => {
 	} );
 };
 
+/**
+ * Determines whether a block pattern has been inserted and if so, records
+ * a track event for it. The recorded event will also reflect whether the
+ * inserted pattern replaced blocks.
+ *
+ * @param {Array} args Data supplied to block insertion or replacement tracking functions.
+ * @returns {string} Pattern name being inserted if available.
+ */
+const trackPatternInsertion = ( args ) => {
+	const meta = find( args, ( arg ) => arg?.patternName );
```

Comment:
`args` is an array so you can use native syntax:

```js
const meta = args.find( ( arg ) => arg?.patternName );
```

Reply:
I skipped this suggestion in https://github.com/Automattic/wp-calypso/pull/46079 as the args were spread into a new object literal when the `blocks_replaced` flag is added. I chose to update the argument name to hopefully avoid confusion instead of updating the other locations that data is passed to this function.